compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/AnnotationDesugar.java

```diff
@@ -628,6 +631,43 @@ private BLangLambdaFunction defineAnnotations(BLangTypeDefinition typeDef, BLang
         return lambdaFunction;
     }
 
+    private BLangLambdaFunction defineTupleAnnotations(BLangTypeDefinition typeDef, BLangPackage pkgNode, SymbolEnv env,
+                                                  PackageID pkgID, BSymbol owner) {
+        BLangFunction function = null;
+        BLangRecordLiteral mapLiteral = null;
+        BLangLambdaFunction lambdaFunction = null;
+
+        if (!typeDef.annAttachments.isEmpty()) {
+            function = defineFunction(typeDef.pos, pkgID, owner);
+            mapLiteral = ASTBuilderUtil.createEmptyRecordLiteral(function.pos, symTable.mapType);
+            addAnnotsToLiteral(typeDef.annAttachments, mapLiteral, function, env, false);
+        }
+        int i = 0;
+        for (BLangSimpleVariable member : ((BLangTupleTypeNode) typeDef.typeNode).memberTypeNodes) {
+            BLangLambdaFunction paramAnnotLambda = defineAnnotations(member.annAttachments, member.pos, pkgNode, env,
+                    pkgID, owner, false);
+            if (paramAnnotLambda != null) {
+                if (function == null) {
+                    function = defineFunction(typeDef.pos, pkgID, owner);
+                    mapLiteral = ASTBuilderUtil.createEmptyRecordLiteral(function.pos, symTable.mapType);
+                }
+
+                addInvocationToLiteral(mapLiteral, FIELD + DOT + i,
+                        member.annAttachments.get(0).pos, paramAnnotLambda);
+            }
+            i++;
+        }
+
+        if (function != null) {
+            if (mapLiteral.fields.isEmpty()) {
+                return null;
+            }
+            lambdaFunction = addReturnAndDefineLambda(function, mapLiteral, pkgNode, env, pkgID, owner);
+        }
+
+        return lambdaFunction;
```

Comment:
Instead of defining a new variable `lambdaFunction` shall we do something like this?
```ballerina
        if (function == null || mapLiteral.fields.isEmpty()) {
            return null;
        }
        return addReturnAndDefineLambda(function, mapLiteral, pkgNode, env, pkgID, owner);
```